SNAP-2886 - executing streaming queries in separate scheduler pool when configured

vatsalmevada opened the pull request:
## Changes proposed in this pull request

Executing streaming queries as part of custom scheduler pool if passed by snappydata.scheduler.pool property.

## Patch testing

manual, scala tests, spark tests

## Other PRs 

https://github.com/SnappyDataInc/snappydata/pull/1248

Commit:
SNAP-2886 - executing streaming queries in separate schedular pool when
configured

suranjan commented on sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala:
```diff
@@ -188,6 +188,17 @@ class StreamExecution(
+        val pool = sparkSession.conf.get("snappydata.scheduler.pool")
+        if(pool == "default" && sparkSession.sparkContext.getPoolForName("streaming").isDefined){
+          sparkSession.sparkContext.setLocalProperty("spark.scheduler.pool", "streaming")
+        } else {
```
else is not required.

vatsalmevada replied:
else is setting any other custom pool configured by user.

suranjan reviewed (approved):
Minor changes. We need to do some concurrency/perf benchmark.
By default the pool will not be defined. We will have that commented for user to enable it.

Commit:
Removing dedicated pool for streaming jobs. Instead client code should
set custom pool in session level config and the same pool will be used
by streaming jobs launched using the same session.

Commit:
Merge branch 'snappy/branch-2.1' into SNAP-2886

Commit:
Merge branch 'snappy/branch-2.1' into SNAP-2886

Commit:
Adding additional existence check as `snappydata.scheduler.pool` config
won't be available while using snappy-spark as a spark distribution for
smart connector mode.
